PowerPointLabs/PowerPointLabs/TimerLab/TimerLabPaneWPF.xaml.cs

```diff
@@ -626,65 +518,119 @@ private void HeightTextBox_LostFocus(object sender, RoutedEventArgs e)
         #endregion
 
         #region Timer Helper
-        private void ReformMarkersIfMissing()
+        private void ReformMissingComponents()
         {
-            int lineMarkerColor = LineMarkerColor();
-            int timeMarkerColor = TimeMarkerColor();
+            bool isTimerBodyRecreated = ReformTimerBodyIfMissing();
+            bool isMarkersRecreated = ReformMarkersIfMissing();
+            bool isSliderRecreated = ReformSliderIfMissing();
+
+            if (isTimerBodyRecreated || isMarkersRecreated || isSliderRecreated)
+            {
+                AdjustZOrder();
+            }
+        }
 
-            // remove current marker
-            Shape lineMarkerGroup = GetLineMarkerGroup();
+        private bool ReformTimerBodyIfMissing()
+        {
+            if (timerBody == null)
+            {
+                AddTimerBody(TimerWidth(), TimerHeight(), 
+                            DefaultTimerLeft(SlideWidth(), TimerWidth()),
+                            DefaultTimerTop(SlideHeight(), TimerHeight()), 
+                            TimerBodyColor());
+                return true;
+            }
+            timerBody.Rotation = 0;
+            return false;
+        }
+
+        private bool ReformMarkersIfMissing()
+        {
+            if (lineMarkerGroup == null || timeMarkerGroup == null)
+            {
+                RecreateMarkers();
```

Comment:
If I'm not wrong, this will reset the markers to the default color. When either one of time or line marker is missing, the other one that exists will be deleted and then recreated with the default color.